crates/metadata-server/src/metric_definitions.rs

```diff
@@ -0,0 +1,166 @@
+    );
+
+    describe_histogram!(
+        METADATA_SERVER_GET_VERSION_DURATION,
+        Unit::Seconds,
+        "Metadata get_version request duration in seconds as measured by the metadata handler"
+    );
+
+    describe_histogram!(
+        METADATA_SERVER_PUT_DURATION,
+        Unit::Seconds,
+        "Metadata put request duration in seconds as measured by the metadata handler"
+    );
+
+    describe_histogram!(
+        METADATA_SERVER_DELETE_DURATION,
+        Unit::Seconds,
+        "Metadata delete request duration in seconds as measured by the metadata handler"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_GET_TOTAL,
+        Unit::Count,
+        "Metadata get request success count as measured by the metadata handler"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_GET_VERSION_TOTAL,
+        Unit::Count,
+        "Metadata get_version success request count as measured by the metadata handler"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_PUT_TOTAL,
+        Unit::Count,
+        "Metadata put request success count as measured by the metadata handler"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_DELETE_TOTAL,
+        Unit::Count,
+        "Metadata delete request success count as measured by the metadata handler"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_EMBEDDED_SENT_MESSAGE_TOTAL,
+        Unit::Count,
+        "Raft Metadata server sent messages count"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_EMBEDDED_RECV_MESSAGE_TOTAL,
+        Unit::Count,
+        "Raft Metadata server received messages count"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_EMBEDDED_SENT_MESSAGE_BYTES,
+        Unit::Bytes,
+        "Raft Metadata server sent messages size in bytes"
+    );
+
+    describe_counter!(
+        METADATA_SERVER_EMBEDDED_RECV_MESSAGE_BYTES,
+        Unit::Bytes,
+        "Raft Metadata server received messages size in bytes"
+    );
+
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_LEADER_ID,
+        "Raft Metadata server know leader id"
+    );
+
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_SNAPSHOT_SIZE_BYTES,
+        Unit::Bytes,
+        "Raft Metadata snapshot size"
+    );
+
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_TERM,
+        Unit::Count,
+        "Raft Metadata raft term number"
+    );
+
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_APPLIED_LSN,
+        Unit::Count,
+        "Raft Metadata raft applied lsn"
+    );
+
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_COMMITTED_LSN,
+        Unit::Count,
+        "Raft Metadata raft committed lsn"
+    );
+
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_FIRST_INDEX,
+        Unit::Count,
+        "Raft Metadata raft first index"
+    );
+    describe_gauge!(
+        METADATA_SERVER_EMBEDDED_LAST_INDEX,
+        Unit::Count,
+        "Raft Metadata raft last index"
```

Comment:
embedded vs. raft vs. replicated? cc. @tillrohrmann

Reply:
I've renamed the metadata server kind to "replicated" in this PR: https://github.com/restatedev/restate/pull/2632.